src/popup/NFTs/NFTs.tsx

```diff
@@ -1,22 +1,26 @@
 import { hooks, Components, NFT as NFTData } from '@reef-chain/react-lib';
 import Uik from '@reef-chain/ui-kit';
 import { reefState } from '@reef-chain/util-lib';
-import React, { useContext, useState } from 'react';
+import React, { useContext, useMemo, useState } from 'react';
 import ReefSigners from '../context/ReefSigners';
 import strings from '../../i18n/locales';
 import { useTheme } from '../context/ThemeContext';
+import SqwidButton from './SqwidButton';
 const { NFTCard, OverlayNFT } = Components;
 
 function NFTs() {
-    const nfts = hooks.useObservableState(reefState.selectedNFTs$);
+    const nftsStatus = hooks.useObservableState(reefState.selectedNFTs_status$);
     const [selectedNFT, setSelectedNFT] = useState<NFTData | undefined>(undefined)
     const { accounts, selectedSigner, provider } = useContext(ReefSigners);
     const { isDarkMode } = useTheme();
+    const nfts = nftsStatus?.data.map((val) => val.data);
```

Comment:
we should just pass nftsStatus.data to render items - and each item has its own status to display balance preloader icon like in mobile app